src/main/java/mSearch/filmlisten/FilmlisteLesen.java

```diff
@@ -119,125 +82,187 @@ private InputStream selectDecompressor(String source, InputStream in) throws Exc
         return in;
     }
 
-    public void readFilmListe(String source, final ListeFilme listeFilme, int days) {
-        Log.sysLog("Liste Filme lesen von: " + source);
+    private void readData(JsonParser jp, ListeFilme listeFilme) throws IOException {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 57 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3776)